examples/audio/melgan_spectrogram_inversion.py

```diff
@@ -0,0 +1,619 @@
+            Conv1D(filters, 3, dilation_rate=3, padding="same"), data_init=False
+        )
+        self.c4 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.c5 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=9, padding="same"), data_init=False
+        )
+        self.c6 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.lrelu = LeakyReLU
+
+    def call(self, input):
+        x1 = self.c1(input)
+        x2 = self.lrelu()(x1)
+        x3 = self.c2(x2)
+        x4 = x3 + input
+
+        x5 = self.lrelu()(x4)
+        x6 = self.c3(x5)
+        x7 = self.lrelu()(x6)
+        x8 = self.c4(x7)
+        x9 = x4 + x8
+
+        x10 = self.lrelu()(x9)
+        x11 = self.c5(x10)
+        x12 = self.lrelu()(x11)
+        x13 = self.c6(x12)
+        x14 = x9 + x13
+
+        return x14
+
+    def get_config(self):
+        config = super(ResidualStack, self).get_config()
+        config.update({"filters": self.filters})
+        return config
+
+
+"""
+Each Convolutional Block uses the dilations offered by the residual stack
+and upsamples the input data by the `upsampling_factor`.
+"""
+
+# Dilated Convolutional Block consisting of the Residual stack
+
+
+class ConvBlock(tf.keras.layers.Layer):
+    """
+  Dilated Convolutional Block with weight normalization.
+
+  input:  conv_dim: int, determines filter size for the block
+          upsampling_factor: int, scale for upsampling
+
+  output: keras.layer, convolutional stack output
+  """
+
+    def __init__(self, conv_dim, upsampling_factor, **kwargs):
+        super().__init__(**kwargs)
+        self.conv_dim = conv_dim
+        self.upsampling_factor = upsampling_factor
+        self.conv_t = addon_layers.WeightNormalization(
+            Conv1DTranspose(self.conv_dim, 16, self.upsampling_factor, padding="same"),
+            data_init=False,
+        )
+        self.lrelu1 = LeakyReLU()
+        self.residual_stack = ResidualStack(self.conv_dim)
+        self.lrelu2 = LeakyReLU()
+
+    def call(self, input):
+        out = self.conv_t(input)
+        out = self.lrelu1(out)
+        out = self.residual_stack(out)
+        out = self.lrelu2(out)
+        return out
+
+    def get_config(self):
+        config = super(ConvBlock, self).get_config()
+        config.update(
+            {"conv_dim": self.conv_dim, "upsampling_factor": self.upsampling_factor}
+        )
+        return config
+
+
+"""
+The discriminator block consists of convolutions and downsampling layers. This block is
+essential for the implementation of the feature matching technique.
+
+Each discriminator outputs a list of feature maps that will be compared during training
+to calculate the feature matching loss.
+"""
+
+
+class DiscriminatorBlock(tf.keras.layers.Layer):
```

Comment:
Make this a Functional model building function

Reply:
Please see the response above